app/lib/actions/katello/katello_applicability/generate_host_applicability.rb

```diff
@@ -0,0 +1,69 @@
+module Actions
+  module Katello
+    module KatelloApplicability
+      class GenerateHostApplicability < Actions::EntryAction
+        input_format do
+          param :host_id, Integer
+        end
+
+        def run
+          content_facet = ::Host.find(input[:host_id]).content_facet
+
+          content_uuids = ::Katello::Rpm.joins("inner join katello_installed_packages on \
+                               katello_rpms.name = katello_installed_packages.name and \
+                               katello_rpms.evr > katello_installed_packages.evr").
+                               joins("INNER JOIN katello_host_installed_packages ON \
+                                     katello_installed_packages.id = katello_host_installed_packages.installed_package_id \
+                                     WHERE katello_host_installed_packages.host_id = #{input[:host_id]}").pluck(:pulp_id)
+
+          consumer_uuids = content_facet.send(applicable_units).pluck("#{content_unit_class.table_name}.pulp_id")
+
+          to_remove = consumer_uuids - content_uuids
+          to_add = content_uuids - consumer_uuids
+
+          ActiveRecord::Base.transaction do
+            insert(to_add, content_facet) unless to_add.blank?
+            remove(to_remove, content_facet) unless to_remove.blank?
+          end
+
+          output[:response] = "content_uuids were #{content_uuids}. " \
+                              "I ran insert(#{to_add}) and remove(#{to_remove})."
+        end
+
+        def resource_locks
+          :link
+        end
+
+        def insert(pulp_ids, content_facet)
+          applicable_ids = content_unit_class.where(:pulp_id => pulp_ids).pluck(:id)
+          unless applicable_ids.empty?
+            inserts = applicable_ids.map { |applicable_id| "(#{applicable_id.to_i}, #{content_facet.id.to_i})" }
+            sql = "INSERT INTO #{content_unit_class.content_facet_association_class.table_name} (#{content_unit_association_id}, content_facet_id) VALUES #{inserts.join(', ')}"
+            ActiveRecord::Base.connection.execute(sql)
+          end
+        end
+
+        def remove(pulp_ids, content_facet)
+          applicable_ids = content_unit_class.where(:pulp_id => pulp_ids).pluck(:id)
+          content_unit_class.content_facet_association_class.where(:content_facet_id => content_facet.id, content_unit_association_id => applicable_ids).delete_all
+        end
+
+        def content_unit_class
+          ::Katello::Rpm
+        end
+
+        def content_unit_association_id
+          "#{content_unit_class.name.demodulize.underscore}_id".to_sym
+        end
+
+        def content_units
+          content_unit_class.name.demodulize.pluralize.underscore.to_sym
+        end
+
+        def applicable_units
+          "applicable_#{content_units}".to_sym
+        end
+      end
+    end
+  end
+end
```

Comment:
This looks like it only does 'packages',  IS there another issue to then calculate errata from this list?

Reply:
or do you want to do it here too?

Reply:
Yeah I was starting with packages first.  I'm okay with doing errata applicability in this PR too, it might not be a big next step.